go/private/rules/helpers.bzl

```diff
@@ -0,0 +1,89 @@
+# Copyright 2017 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+load("@io_bazel_rules_go//go/private:common.bzl",
+    "go_importpath",
+    "sets",
+)
+load("@io_bazel_rules_go//go/private:providers.bzl",
+    "GoLibrary",
+    "GoSource",
+    "GoArchive",
+    "GoAspectProviders",
+)
+
+def new_go_library(ctx, resolver=None, importable=True, **kwargs):
```

Comment:
Need documentation. In particular, what is the type of `resolver`, and what is it supposed to do?

Reply:
Will document in a follow up cl, the interface changes the the toolchain and stdlib cleanup that comes next.